markdown_it/tree.py

```diff
@@ -2,12 +2,30 @@
 
 This module is not part of upstream JavaScript markdown-it.
 """
-from typing import NamedTuple, Sequence, Tuple, Dict, List, Optional, Any
+from typing import (
+    NamedTuple,
+    Sequence,
+    Tuple,
+    Dict,
+    List,
+    Optional,
+    Any,
+    TypeVar,
+    Type,
+)
 
 from .token import Token
 from .utils import _removesuffix
 
 
+class _NesterTokens(NamedTuple):
+    opening: Token
+    closing: Token
+
+
+_T = TypeVar("_T", bound="SyntaxTreeNode")
```

Comment:
why `_T`? it would be better from a readability perspective to use something like `NodeType`

Reply:
It seems like its a convention to name `TypeVar`s as `T` (or if private, `_T`). I think this is because its short, and because the same `TypeVar` could be reused in various contexts having a different meaning in each. Here setting the `bound` sort of limits any reasonable reusability though.

That being said, I don't oppose renaming at all if you think thats better :smile:

Reply:
yes I think so thanks

Reply:
changed to `_NodeType`